src/pydicom/pixels/utils.py

```diff
@@ -1324,6 +1362,40 @@ def pack_bits(arr: "np.ndarray", pad: bool = True) -> bytes:
     :dcm:`Section 8.1.1<part05/chapter_8.html#sect_8.1.1>` and
     :dcm:`Annex D<part05/chapter_D.html>`
     """
+    if isinstance(arr, bytes | bytearray):
+        if len(arr) == 0:
+            return b""
+
+        if HAVE_NP:
+            # Form an array because the numpy implementation is more efficient
+            arr = np.frombuffer(arr, dtype=np.uint8)
+        else:
+            # Fallback for when numpy is not installed (slower)
```

Comment:
Mostly just a stylistic choice, but you could `return pack_bits(np.frombuffer(arr, dtype=np.uint8)` to remove the need for the `else` indent.

Reply:
Addressed in [e9ea3de](https://github.com/pydicom/pydicom/pull/2216/commits/e9ea3deb538f6786f5a80e0e826d7aaf7a40678c)